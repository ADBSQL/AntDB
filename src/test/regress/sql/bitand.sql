set grammar to oracle;
select bitand(1,3) from dual;
select bitand('1',3) from dual;
select bitand(127,3.9456) from dual;
select bitand('1.4999',1.4999) from dual;
select bitand(-1,-1) from dual;
select bitand('-3',2) from dual;
select bitand(127.99,'128.99') from dual;
select bitand('-127','128') from dual;
select bitand(exp(2),exp(2)) from dual;
select bitand(to_char(4),0) from dual;
select bitand(3*3,2*4) from dual;
select bitand('s',0) from dual;
select bitand('',0) from dual;
select bitand('',null) from dual;
select bitand(null,2) from dual;
