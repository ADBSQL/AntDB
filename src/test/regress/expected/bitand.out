set grammar to oracle;
select bitand(1,3) from dual;
 BITAND(1,3) 
-------------
           1
(1 row)

select bitand('1',3) from dual;
 BITAND('1',3) 
---------------
             1
(1 row)

select bitand(127,3.9456) from dual;
 BITAND(127,3.9456) 
--------------------
                  3
(1 row)

select bitand('1.4999',1.4999) from dual;
 BITAND('1.4999',1.4999) 
-------------------------
                       1
(1 row)

select bitand(-1,-1) from dual;
 BITAND(-1,-1) 
---------------
            -1
(1 row)

select bitand('-3',2) from dual;
 BITAND('-3',2) 
----------------
              0
(1 row)

select bitand(127.99,'128.99') from dual;
 BITAND(127.99,'128.99') 
-------------------------
                       0
(1 row)

select bitand('-127','128') from dual;
 BITAND('-127','128') 
----------------------
                  128
(1 row)

select bitand(exp(2),exp(2)) from dual;
 BITAND(EXP(2),EXP(2)) 
-----------------------
                     7
(1 row)

select bitand(to_char(4),0) from dual;
 BITAND(TO_CHAR(4),0) 
----------------------
                    0
(1 row)

select bitand(3*3,2*4) from dual;
 BITAND(3*3,2*4) 
-----------------
               8
(1 row)

select bitand('s',0) from dual;
ERROR:  invalid input syntax for type numeric: "s"
LINE 1: select bitand('s',0) from dual;
                      ^
select bitand('',0) from dual;
 BITAND('',0) 
--------------
             
(1 row)

select bitand('',null) from dual;
 BITAND('',NULL) 
-----------------
                
(1 row)

select bitand(null,2) from dual;
 BITAND(NULL,2) 
----------------
               
(1 row)

